app/client/src/contexts/MapHighlight.tsx

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext, useMemo, useState } from 'react';
+import type { ReactNode } from 'react';
+import Graphic from '@arcgis/core/Graphic';
+
+type State = {
+  highlightedGraphic: Graphic | null;
+  selectedGraphic: Graphic | null;
+  setHighlightedGraphic: (graphic: Graphic | null) => void;
+  setSelectedGraphic: (graphic: Graphic | null) => void;
```

Comment:
```suggestion
  highlightedGraphic: Graphic | null;
  selectedGraphic: Graphic | null;
  setHighlightedGraphic: Dispatch<SetStateAction<Graphic | null>>;
  setSelectedGraphic: Dispatch<SetStateAction<Graphic | null>>;
```

This should be using the React types.